### DISCUSSION TITLE
how to retain static properties on a layout component

### DISCUSSION THREAD

**Discussion by stefanprobst** [mdx-js/mdx#1825] [Q&A]
when wrapping a mdx document in a layout component via default export, how can i retain static properties defined on that layout component function?

example usecase: https://github.com/stefanprobst/issue-mdx-layout-props

in the above example, the mdx document default-exports a `Layout` component with a `getPageWrapper` property. how can i retain that `getPageWrapper` property on the default export after compilation (i.e. move it from `MDXLayout` to `MDXContent`)?

**Comment by wooorm** [comment]
Why have this written in MDX though? Maybe this is distilled down, but it looks like only JavaScript, so why not have that in JavaScript?
